knapsack-highs/main.py

```diff
@@ -2,134 +2,84 @@
 Template for working with highspy.
 """
 
-import argparse
-import json
-import sys
 import time
 from importlib.metadata import version
-from typing import Any
 
 import highspy
+import nextmv
 
 
 def main() -> None:
-    """Entry point for the template."""
+    """Entry point for the program."""
 
-    parser = argparse.ArgumentParser(description="Solve problems with highspy.")
-    parser.add_argument(
-        "-input",
-        default="",
-        help="Path to input file. Default is stdin.",
+    options = nextmv.Options(
+        nextmv.Parameter("input", str, "", "Path to input file. Default is stdin.", False),
+        nextmv.Parameter("output", str, "", "Path to output file. Default is stdout.", False),
+        nextmv.Parameter("duration", int, 30, "Max runtime duration (in seconds).", False),
     )
-    parser.add_argument(
-        "-output",
-        default="",
-        help="Path to output file. Default is stdout.",
-    )
-    parser.add_argument(
-        "-duration",
-        default=30,
-        help="Max runtime duration (in seconds). Default is 30.",
-        type=int,
-    )
-    args = parser.parse_args()
 
-    # Read input data, solve the problem and write the solution.
-    input_data = read_input(args.input)
+    input = nextmv.load_local(options=options, path=options.input)
 
-    log("Solving knapsack problem:")
-    log(f"  - items: {len(input_data.get('items', []))}")
-    log(f"  - capacity: {input_data.get('weight_capacity', 0)}")
-    log(f"  - max duration: {args.duration} seconds")
+    nextmv.log("Solving knapsack problem:")
+    nextmv.log(f"  - items: {len(input.data.get('items', []))}")
+    nextmv.log(f"  - capacity: {input.data.get('weight_capacity', 0)}")
 
-    solution = solve(input_data, args.duration)
-    write_output(args.output, solution)
+    output = solve(input, options)
+    nextmv.write_local(output, path=options.output)
 
 
-def solve(input_data: dict[str, Any], duration: int) -> dict[str, Any]:
+def solve(input: nextmv.Input, options: nextmv.Options) -> nextmv.Output:
     """Solves the given problem and returns the solution."""
 
-    start = time.time()
+    start_time = time.time()
+    nextmv.redirect_stdout()  # Solver chatter is logged to stderr.
 
     # Creates the solver.
     solver = highspy.Highs()
-    solver.silent()
-    solver.setOptionValue("time_limit", duration)
+    solver.silent()  # Solver output always logged to stdout, ignores redirect.
```

Comment:
```suggestion
    solver.silent()  # Solver output ignores stdout redirect, silence it.
```

Just a suggestion for clarity. :shrug: